lib/pos_random_seed.go

```diff
@@ -64,17 +64,27 @@ func (randomSeedHash *RandomSeedHash) isEmpty() bool {
 // UTXO VIEW UTILS
 //
 
-func (bav *UtxoView) GenerateRandomSeedSignature(signerPrivateKey *bls.PrivateKey) (*bls.Signature, error) {
+func (bav *UtxoView) GenerateCurrentRandomSeedSignature(signerPrivateKey *bls.PrivateKey) (*bls.Signature, error) {
 	// This function generates a RandomSeedSignature by signing the CurrentRandomSeedHash
 	// with the provided bls.PrivateKey. This signature is deterministic: given the same
 	// CurrentRandomSeedHash and bls.PrivateKey, the same signature will always be generated.
 	currentRandomSeedHash, err := bav.GetCurrentRandomSeedHash()
 	if err != nil {
-		return nil, errors.Wrapf(err, "UtxoView.GenerateRandomSeedSignature: problem retrieving CurrentRandomSeedHash: ")
+		return nil, errors.Wrapf(err, "UtxoView.GenerateCurrentRandomSeedSignature: problem retrieving CurrentRandomSeedHash: ")
 	}
-	randomSeedSignature, err := signerPrivateKey.Sign(currentRandomSeedHash[:])
+	randomSeedSignature, err := GenerateRandomSeedSignatureFromRandomSeedHash(signerPrivateKey, currentRandomSeedHash)
 	if err != nil {
-		return nil, errors.Wrapf(err, "UtxoView.GenerateRandomSeedSignature: problem signing CurrentRandomSeedHash: ")
+		return nil, errors.Wrapf(err, "UtxoView.GenerateCurrentRandomSeedSignature: problem generating RandomSeedSignature: ")
+	}
+	return randomSeedSignature, nil
+}
+
+func GenerateRandomSeedSignatureFromRandomSeedHash(
```

Comment:
Just a signature. May as well simplify the name to describe what it does
```suggestion
func SignRandomSeedHash(
```